pkg/scheduler/core/generic_scheduler.go

```diff
@@ -361,14 +368,45 @@ func (g *genericScheduler) findNodesThatPassExtenders(pod *v1.Pod, feasibleNodes
 			return nil, err
 		}
 
+		for failedNodeName, failedMsg := range failedAndUnresolvableMap {
+			if _, found := statuses[failedNodeName]; !found {
+				statuses[failedNodeName] = framework.NewStatus(framework.UnschedulableAndUnresolvable, failedMsg)
+			} else {
+				if statuses[failedNodeName].Code() != framework.UnschedulableAndUnresolvable {
+					// UnschedulableAndUnresolvable takes precedence over Unschedulable
+					statuses[failedNodeName] = framework.NewStatus(framework.UnschedulableAndUnresolvable, statuses[failedNodeName].Reasons()...)
+				}
+				statuses[failedNodeName].AppendReason(failedMsg)
+			}
+		}
+
 		for failedNodeName, failedMsg := range failedMap {
+			if _, found := failedAndUnresolvableMap[failedNodeName]; found {
+				// failedAndUnresolvableMap takes precedence over failedMap
+				continue
+			}
 			if _, found := statuses[failedNodeName]; !found {
 				statuses[failedNodeName] = framework.NewStatus(framework.Unschedulable, failedMsg)
 			} else {
 				statuses[failedNodeName].AppendReason(failedMsg)
 			}
 		}
-		feasibleNodes = feasibleList
+
+		prevFailedButResolvableNodes = make(map[string]*v1.Node)
+		for _, node := range feasibleNodes {
+			if status, ok := statuses[node.Name]; ok && status.Code() == framework.Unschedulable {
+				prevFailedButResolvableNodes[node.Name] = node
+			}
+		}
+
+		// Failed but unresolvable nodes may be passed to current extender, we
+		// should exclude them.
```

Comment:
Same as previous commit, the `feasibleNodes` returned shouldn't contain any failed node (with any failure reason).

Reply:
done